libs/cardano-ledger-api/src/Cardano/Ledger/Api/Scripts/ExUnits.hs

```diff
@@ -51,44 +38,55 @@ import Cardano.Ledger.Alonzo.TxWits (
   unRedeemers,
   unTxDats,
  )
-import Cardano.Ledger.Alonzo.UTxO (AlonzoScriptsNeeded (..))
-import Cardano.Ledger.BaseTypes (StrictMaybe (..))
-import Cardano.Ledger.Core hiding (TranslationError)
-import Cardano.Ledger.Plutus.Data (Data, Datum (..), binaryDataToData, getPlutusData)
-import Cardano.Ledger.Plutus.Language (BinaryPlutus (..), Language (..), Plutus (..), SLanguage (..))
+import Cardano.Ledger.Alonzo.UTxO (AlonzoScriptsNeeded (..), getAlonzoSpendingTxIn)
+import Cardano.Ledger.BaseTypes (StrictMaybe (..), pvMajor)
+import Cardano.Ledger.Core
+import Cardano.Ledger.Plutus.Data (Datum (..), binaryDataToData, getPlutusData)
+import Cardano.Ledger.Plutus.Evaluate (
+  PlutusDatums (..),
+  PlutusWithContext (..),
+  evaluatePlutusWithContext,
+ )
+import Cardano.Ledger.Plutus.ExUnits
+import Cardano.Ledger.Plutus.Language (Language (..))
+import Cardano.Ledger.Plutus.TxInfo (exBudgetToExUnits)
 import Cardano.Ledger.TxIn (TxIn)
 import Cardano.Ledger.UTxO (EraUTxO (..), ScriptsProvided (..), UTxO (..))
 import Cardano.Slotting.EpochInfo.API (EpochInfo)
 import Cardano.Slotting.Time (SystemStart)
-import Data.Array (Array, bounds, (!))
+import Control.Monad (forM)
 import Data.Map.Strict (Map)
 import qualified Data.Map.Strict as Map
-import Data.Maybe (mapMaybe)
-import qualified Data.Set as Set
 import Data.Text (Text)
 import Lens.Micro
-import qualified PlutusLedgerApi.Common as Plutus
-import qualified PlutusLedgerApi.V1 as PV1
-import qualified PlutusLedgerApi.V2 as PV2
-import qualified PlutusLedgerApi.V3 as PV3
+import qualified PlutusLedgerApi.Common as P
 
--- | Script failures that can be returned by 'evaluateTransactionExecutionUnits'.
+-- | Script failures that can be returned by 'evalTxExUnitsWithLogs'.
 data TransactionScriptFailure era
-  = -- | A redeemer was supplied that does not point to a
-    --  valid plutus evaluation site in the given transaction.
-    RedeemerNotNeeded !RdmrPtr !(ScriptHash (EraCrypto era))
-  | -- | A redeemer was supplied which points to a script hash which
+  = -- | A redeemer was supplied which points to a script hash which
     -- we cannot connect to a Plutus script.
     RedeemerPointsToUnknownScriptHash !RdmrPtr
-  | -- | Missing redeemer. The first parameter is the redeemer pointer which cannot be resolved,
+  | -- | Missing redeemer. The first parameter is the ,
```

Comment:
the what?

Reply:
Thank you for catching this. I meant to remove the whole sentence, since I moved the documentation to the actual parameters (haddock nowadays supports this)